docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md

```diff
@@ -0,0 +1,177 @@
+
+### [Bitnami Sealed Secrets](https://github.com/bitnami-labs/sealed-secrets)
+
+Sealed secrets use asymmetric encryption to encrypt secrets. Kubernetes controller generates key pair(private-public) and stores the private key in etcd as Kubernetes secret. Developers use [Kubeseal CLI](https://github.com/bitnami-labs/sealed-secrets#public-key--certificate) to seal secrets before committing to the git repo.
+
+- [Support automatic key rotation](https://github.com/bitnami-labs/sealed-secrets#sealing-key-renewal) for the private key and can be used to enforce re-encryption of secrets
+  - Due to [automatic renewal of the sealing key](https://github.com/bitnami-labs/sealed-secrets#sealing-key-renewal), the key needs to be prefetched from the cluster or cluster set up to store the sealing key on renewal in a secondary location
+- Multi-tenancy support at the namespace level can be enforced by the controller
+- When sealing secrets developers needs a connection to the cluster control plane to fetch the public key or the public key has to be explicitly shared with the developer
+- Potential risk of secret checked into a git repo
+- If the private key in the cluster is lost for some reason all secrets need to be re-encrypted followed by a new key-pair generation
+- Does not scale with Multi-cluster here every cluster will require a controller having its own key pair
+- Secrets are stored encrypted in the gitops repository, if the private encryption key is leaked, all secrets can be decrypted
+
+### [Mozilla SOPS](https://github.com/mozilla/sops)
+
+SOPS: Secrets OPerationS is an encryption tool that supports YAML, JSON, ENV, INI, and BINARY formats and encrypts with AWS KMS, GCP KMS, Azure Key Vault, age, and PGP and is not just limited to Kubernetes. It supports integration with some common key management systems including Azure Key Vault. Where the key management system is used to store the encryption key for encrypting secrets and not the actual secrets.
+
+- [Flux](https://toolkit.fluxcd.io/guides/mozilla-sops/#azure) has better support for SOPS with cluster-side decryption.
+- An added layer of security as the private key used for decryption is protected in an external key vault.
+- Needs plugin to work with Helm([Helm Secrets](https://github.com/jkroepke/helm-secrets)) and Kustomization ([KSOPS](https://github.com/viaduct-ai/kustomize-sops))([kustomize-sopssecretgenerator](https://github.com/goabout/kustomize-sopssecretgenerator))
+- Does not scale with larger teams as each developer has to encrypt the secrets
+- The public key is sufficient for creating brand new files. The secret key is required for decrypting and editing existing files because SOPS computes a MAC on all values.  When using the public key solely to add or remove a field, the whole file should be deleted and recreated.
+
+## 2. Reference to secrets stored in an external key vault (recommended)
```

Comment:
Consider adding a sentence that also indicates this is the recommended solution following the bulleted list that introduces these above (will repeat this comment up there.)

Reply:
You'd like to add that only at the top, right? This is just a duplicate comment?

Reply:
Yes.